File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/ActivityRepositoryInterface.java
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository;

import androidx.lifecycle.LiveData;

import java.util.List;

import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.Activity;

public interface ActivityRepositoryInterface {
    void insertActivity(String title, String description, long date);

    LiveData<List<Activity>> getAllUndoneActivity();

    LiveData<List<Activity>> getTodayActivity();

    void updateActivity(String idAct);
}

File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/ActivityRepositoryInterface.kt
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository

import androidx.lifecycle.LiveData
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.Activity


interface ActivityRepositoryInterface {
    suspend fun insertActivity(title: String, description: String, date: Long)
    suspend fun updateActivity(idAct: String)
    fun getTodayActivity(): LiveData<List<Activity>>
    fun getAllUndoneActivity(): LiveData<List<Activity>>
}